### ISSUE TITLE
macbook pro 2014 no changes after apply offset for CPU

### ISSUE THREAD

**Issue by Askhento** [sicreative/VoltageShift#97]
Is it possible to run the program on my machine?

```bash
➜  voltageshift_1.25 ./voltageshift offset -50
--------------------------------------------------------------------------
VoltageShift offset Tool
--------------------------------------------------------------------------
Before CPU voltageoffset: 0mv
--------------------------------------------------------------------------
After CPU voltageoffset: 0mv
--------------------------------------------------------------------------

➜  voltageshift_1.25 ./voltageshift info
------------------------------------------------------
   VoltageShift Info Tool
------------------------------------------------------
CPU voltage offset: 0mv
GPU voltage offset: 0mv
CPU Cache voltage offset: 0mv
System Agency offset: 0mv
Analogy I/O: 0mv
Digital I/O: 0mv
CPU BaseFreq: 2500, CPU MaxFreq(1/2/4): 3700/3600/3500 (mhz)OC_Locked  PL1: 100W PL2: 125W
CPU Freq: 0.8ghz, Voltage: 0.6753v, Power:pkg 10.77w /core 2.95w,Temp: 50 c

```

OS : Catalina 10.15.7 (19H2026)

**Comment by teqtic** [comment]
Which firmware version do you have installed (System report -> hardware -> System Firmware Version)? Did the laptop ever run Big Sur? If it did, the firmware update would have locked the register to which the undervolts apply. If this is the case you will need to modify your firmware with a physical programmer as explained below:

https://github.com/IsmaelHG/UndervoltPatchMacBook

I just did this and can confirm it works with version 478.0.0.0.

**Comment by Vasllo** [comment]
> Which firmware version do you have installed (System report -> hardware -> System Firmware Version)? Did the laptop ever run Big Sur? If it did, the firmware update would have locked the register to which the undervolts apply. If this is the case you will need to modify your firmware with a physical programmer as explained below:
> 
> https://github.com/IsmaelHG/UndervoltPatchMacBook
> 
> I just did this and can confirm it works with version 478.0.0.0.

Hi there, I'm having the same issue with the same model and wanted to confirm your suspicions, it's exactly what you've mentioned. Mine had Big Sur before using OpenCore Legacy Patcher to upgrade to Sonoma. The firmware version is the same as yours, but I get the OC_Locked message. Thankfully the PL1 and PL2 settings work and that's enough for me, I don't think it's worth going into that rabbit hole for 50mV of undervolt. Thanks for the software! I tried Volta, but it would keep disabling Turbo Boost and wouldn't open after restarting.

**Comment by teqtic** [comment]
> > Which firmware version do you have installed (System report -> hardware -> System Firmware Version)? Did the laptop ever run Big Sur? If it did, the firmware update would have locked the register to which the undervolts apply. If this is the case you will need to modify your firmware with a physical programmer as explained below:
> > https://github.com/IsmaelHG/UndervoltPatchMacBook
> > I just did this and can confirm it works with version 478.0.0.0.
> 
> Hi there, I'm having the same issue with the same model and wanted to confirm your suspicions, it's exactly what you've mentioned. Mine had Big Sur before using OpenCore Legacy Patcher to upgrade to Sonoma. The firmware version is the same as yours, but I get the OC_Locked message. Thankfully the PL1 and PL2 settings work and that's enough for me, I don't think it's worth going into that rabbit hole for 50mV of undervolt. Thanks for the software! I tried Volta, but it would keep disabling Turbo Boost and wouldn't open after restarting.

Yes, and Volta won't work at applying the undervolt either, without patching the firmware with a physical programmer.

**Comment by Vasllo** [comment]
> > > Which firmware version do you have installed (System report -> hardware -> System Firmware Version)? Did the laptop ever run Big Sur? If it did, the firmware update would have locked the register to which the undervolts apply. If this is the case you will need to modify your firmware with a physical programmer as explained below:
> > > https://github.com/IsmaelHG/UndervoltPatchMacBook
> > > I just did this and can confirm it works with version 478.0.0.0.
> > 
> > 
> > Hi there, I'm having the same issue with the same model and wanted to confirm your suspicions, it's exactly what you've mentioned. Mine had Big Sur before using OpenCore Legacy Patcher to upgrade to Sonoma. The firmware version is the same as yours, but I get the OC_Locked message. Thankfully the PL1 and PL2 settings work and that's enough for me, I don't think it's worth going into that rabbit hole for 50mV of undervolt. Thanks for the software! I tried Volta, but it would keep disabling Turbo Boost and wouldn't open after restarting.
> 
> Yes, and Volta won't work at applying the undervolt either, without patching the firmware with a physical programmer.

Yes, but the main issue I was having with it was that it was disabling Turbo Boost on its own from time to time, also wouldn't load by itself and would require to open, close and open it again to work every boot.
